### PR TITLE
backend/feat/parcel/send fcm to parties in more cases

### PR DESCRIPTION
**Pull request by ErasedSoul**
Send fcm to parties in more cases -> 
1. Ride Assigned
2. Booking Cancellation
3. Booking Reallocation
4. Fix Parcel Ride end fcm not coming to other parties

Fix race condition in booking [Moved to other pr.] [https://github.com/nammayatri/nammayatri/pull/8844](https://github.com/nammayatri/nammayatri/pull/8844)

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by khuzema786 — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Tools/Notifications.hs [hunk 1]
```diff
@@ -1065,7 +1070,7 @@ getAllOtherRelatedPartyPersons :: ServiceFlow m r => SRB.Booking -> m [Person]
 getAllOtherRelatedPartyPersons booking = do
   case booking.tripCategory of
     Just (Trip.Delivery _) -> do
-      allBookingParties <- QBPL.findAllActiveByBookingId booking.id
+      allBookingParties <- QBPL.findAllByBookingId booking.id
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Tools/Notifications.hs:1073]: Why this change ?

Comment (ErasedSoul) [reply Backend/app/rider-platform/rider-app/Main/src/Tools/Notifications.hs:1073]: due to end ride case .. 
End ride fcm was not coming as all became inactive

Comment (khuzema786) [reply Backend/app/rider-platform/rider-app/Main/src/Tools/Notifications.hs:1073]: We should make them inActive after triggering all FCM and everything

**Review by khuzema786 — APPROVED** [review]

**Review by khuzema786 — APPROVED** [review]
